unblob/processing.py

```diff
@@ -725,28 +731,46 @@ def calculate_entropy(path: Path) -> EntropyReport:
         max_limit=1024 * 1024,
     )
 
-    entropy_sum = 0.0
     with File.from_path(path) as file:
         for chunk in iterate_file(file, 0, file_size, buffer_size=block_size):
-            entropy = mt.shannon_entropy(chunk)
-            entropy_percentage = round(entropy / 8 * 100, 2)
-            percentages.append(entropy_percentage)
-            entropy_sum += entropy * len(chunk)
-
-    report = EntropyReport(
-        percentages=percentages,
-        block_size=block_size,
-        mean=entropy_sum / file_size / 8 * 100,
+            shannon_entropy = mt.shannon_entropy(chunk)
+            shannon_entropy_percentage = round(shannon_entropy / 8 * 100, 2)
+            shannon_percentages.append(shannon_entropy_percentage)
+
+            chi_square_probability = mt.chi_square_probability(chunk)
+            chisquare_probability_percentage = round(chi_square_probability * 100, 2)
+            chi_square_percentages.append(chisquare_probability_percentage)
+
+    report = RandomnessReport(
+        shannon=RandomnessMeasurements(
+            percentages=shannon_percentages,
+            block_size=block_size,
+            mean=mean(shannon_percentages),
```

Comment:
I think the original calculation resulted in different values for means, as the naive mean ignoring file size can be highly biased:
if the file size is 1025, it is processed as 2 chunks:
- 1024 bytes
- 1 bytes

The 1 byte entropy percentage should be 100%, which pushes the overall `mean` higher than it really is for the whole file.  We can somewhat counteract this bias by weighting the values with the size of the chunk.

The above is valid for `chi_square` as well.

Reply:
Fixed.